components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -5269,4 +5301,24 @@ private void setUserNameWithDomain(Map<String, String> userClaimValues, Map<Stri
                     .getDomainQualifiedUsername()));
         }
     }
+
+    private boolean isUserContains(org.wso2.carbon.user.core.common.User coreUser, String user) throws
+            org.wso2.carbon.user.core.UserStoreException {
+
+        if (isLoginIdentifiersEnabled() &&
+                StringUtils.isNotBlank(getPrimaryLoginIdentifierClaim())) {
+            String primaryLoginIdentifier = carbonUM.getUserClaimValue(coreUser.getUsername(),
+                    getPrimaryLoginIdentifierClaim(), null);
+            if (StringUtils.isBlank(primaryLoginIdentifier)) {
+                return user.equalsIgnoreCase(coreUser.getUsername().indexOf(UserCoreConstants.DOMAIN_SEPARATOR) > 0
+                        ? coreUser.getUsername().split(UserCoreConstants.DOMAIN_SEPARATOR)[1] : coreUser.getUsername());
```

Comment:
avoid using ternary operator as it reduces code readability

Reply:
please check other places as well

Reply:
fixed with 11edd4759770f28f9932335ec375ef8a5e3f8846